src/sentry/static/sentry/app/views/organizationEventsV2/data.jsx

```diff
@@ -69,18 +70,23 @@ export const ALL_VIEWS = deepFreeze([
 export const SPECIAL_FIELDS = {
   event: {
     fields: ['title', 'id', 'project.name'],
-    renderFunc: (data, {organization}) => (
-      <Container>
-        <Link
-          css={overflowEllipsis}
-          to={`/organizations/${organization.slug}/projects/${
-            data['project.name']
-          }/events/${data.id}/`}
-        >
-          {data.title}
-        </Link>
-      </Container>
-    ),
+    renderFunc: (data, {organization, location}) => {
+      const newQuery = qs.stringify({
+        ...location.query,
+        eventSlug: `${data['project.name']}:${data.id}`,
+      });
+      return (
+        <Container>
+          <Link
+            css={overflowEllipsis}
+            to={`/organizations/${organization.slug}/events/?${newQuery}`}
```

Comment:
Instead of making our own query string, we could pass `to` an object with `pathname` and `query` keys.